inference-engine/samples/speech_sample/main.cpp

```diff
@@ -633,7 +505,18 @@ int main(int argc, char* argv[]) {
             return 0;
         }
 
-        std::vector<std::string> inputArkFiles;
+        BaseFile* file;
+        BaseFile* fileOutput;
+        auto extInputFile = fileExt(FLAGS_i);
+        if (extInputFile == "ark") {
+            file = new ArkFile();
```

Comment:
Are we using "file" outside of this block? If no then you don't need to allocate it in the heap

Reply:
Now allocate on stack.